hs_core/management/commands/remove_unreferenced_files.py

```diff
@@ -0,0 +1,95 @@
+
+        istorage = IrodsStorage()
+
+        def list_files_recursively(folder_path):
+            try:
+                folders, files, _ = istorage.listdir(folder_path)
+            except SessionException as ex:
+                print(f"Failed to list files in folder path {folder_path}: {ex.stderr}")
+                return []
+            files = [f"{folder_path}/{f}" for f in files]
+            for folder in folders:
+                sub_folder_path = f"{folder_path}/{folder}"
+                try:
+                    subfolders, subfiles, _ = istorage.listdir(sub_folder_path)
+                except SessionException as ex:
+                    print(f"Failed to list files in subfolder path {sub_folder_path}: {ex.stderr}")
+                    subfiles = []
+                    subfolders = []
+                files += ([f"{sub_folder_path}/{f}" for f in subfiles])
+                for subfolder in subfolders:
+                    files += list_files_recursively(f"{sub_folder_path}/{subfolder}")
+            return files
+
+        def special_file(f):
+            if f.endswith("_meta.xml") or f.endswith("_resmap.xml") or f.endswith("_schema.json"):
+                return True
+            return False
+
+        total_resources = resources.count()
+        current_resource_count = 1
+        resources_with_dangling_rf = []
+        resources_with_missing_rf = []
+        for resource in resources.iterator():
+            print(f"{current_resource_count}/{total_resources}")
+            current_resource_count = current_resource_count + 1
+            irods_files = list_files_recursively(resource.file_path)
+            irods_files = [f for f in irods_files if not special_file(f)]
+            res_files = ResourceFile.objects.filter(object_id=resource.id)
+            res_files_with_no_file = res_files.exclude(resource_file__in=irods_files).values_list('resource_file',
+                                                                                                  flat=True)
+            if res_files_with_no_file:
+                # print("Dangline resource files")
+                resources_with_dangling_rf.append(resource.short_id)
+                # res_files_with_no_file.delete()
+                # print(" ".join(res_files_with_no_file))
+            matched_values = res_files.filter(resource_file__in=irods_files).values_list('resource_file', flat=True)
+            unreferenced_irods_files = [f for f in irods_files if f not in list(matched_values)]
+            if unreferenced_irods_files:
+                resources_with_missing_rf.append(resource.short_id)
+                print("Unreferenced irods files")
+                print(" ".join(unreferenced_irods_files))
+                if not dryrun:
+                    for file in unreferenced_irods_files:
+                        try:
+                            istorage.delete(file)
+                        except SessionException as ex:
```

Comment:
I wonder if there are any other exceptions this could throw...I guess SessionException is likely adequate I just don't want to get stuck with this command failing halfway through and not continuing at game time. But it's mostly me being paranoid